### PR TITLE
Integration tests on Iron

### PR DESCRIPTION
**Pull request by VinDp**
This PR is a manual backport of #17, since: 

- some lines from the `ros2` branch were missing in the `iron` branch
- the `test_description.py` was searching for urdfs that do not exist in this branch of the repo, since the description refactoring only happened on rolling

With these minimal changes both tests (gz and description) seem to work.

### CODE REVIEW

File: ur_simulation_gz/test/test_description.py
Comment (fmauch) [inline ur_simulation_gz/test/test_description.py:1]: The test_description script doesn't make too much sense here, in my opinion. For Iron we don't have a custom description in this repo.

File: ur_simulation_gz/package.xml [hunk 1]
```diff
@@ -29,6 +29,12 @@
   <exec_depend>ros_gz_sim</exec_depend>
   <exec_depend>xacro</exec_depend>
 
+  <test_depend>ament_cmake_pytest</test_depend>
+  <test_depend>launch_testing_ament_cmake</test_depend>
+  <test_depend>launch_testing_ros</test_depend>
+  <test_depend>liburdfdom-tools</test_depend>
+  <test_depend>xacro</test_depend>
```
Comment (fmauch) [inline ur_simulation_gz/package.xml]: Probably not needed when the test_description test is removed.

File: ur_simulation_gz/CMakeLists.txt [hunk 2]
```diff
@@ -3,8 +3,28 @@ project(ur_simulation_gz)
...
+  find_package(launch_testing_ament_cmake)
+  find_package(ament_cmake_pytest REQUIRED)
+
+  if(${UR_SIM_INTEGRATION_TESTS})
+    add_launch_test(test/test_gz.py
+      TIMEOUT
+        180
+    )
+  endif()
+  ament_add_pytest_test(description test/test_description.py)
```
Comment (fmauch) [inline ur_simulation_gz/CMakeLists.txt]: This could be removed with the test altogether.

**Review by fmauch — CHANGES_REQUESTED** [review]

File: ur_simulation_gz/test/test_description.py
Comment (VinDp) [reply ur_simulation_gz/test/test_description.py:1]: Yeah, it makes sense. I will remove it.

**Review by fmauch — APPROVED** [review]

**Comment by VinDp** [comment]
@Mergifyio backport humble

**Pull request merged**